pygmt/datasets/tile_map.py

```diff
@@ -0,0 +1,133 @@
+:class:`xarray.DataArray`.
+"""
+
+try:
+    import contextily
+except ImportError:
+    contextily = None
+
+import numpy as np
+import xarray as xr
+
+__doctest_requires__ = {("load_tile_map"): ["contextily"]}
+
+
+def load_tile_map(region, zoom="auto", source=None, lonlat=True, wait=0, max_retries=2):
+    """
+    Load a georeferenced raster basemap from XYZ tile providers.
+
+    The tiles that compose the map are merged and georeferenced into an
+    :class:`xarray.DataArray` image with 3 bands (RGB). Note that the returned
+    image is in a Spherical Mercator (EPSG:3857) coordinate reference system.
+
+    Parameters
+    ----------
+    region : list
+        The bounding box of the map in the form of a list [*xmin*, *xmax*,
+        *ymin*, *ymax*]. These coordinates should be in longitude/latitude if
+        ``lonlat=True`` or Spherical Mercator (EPSG:3857) if ``lonlat=False``.
+
+    zoom : int
+        Optional. Level of detail. [Default is 'auto'].
+
+    source : xyzservices.TileProvider or str
+        Optional. The tile source: web tile provider or path to a local file.
+        The web tile provider can be in the form of a
+        :class:`xyzservices.TileProvider` object or a URL. The placeholders for
+        the XYZ in the URL need to be {x}, {y}, {z}, respectively. For local
+        file paths, the file is read with :doc:`rasterio <rasterio:index>` and
+        all bands are loaded into the basemap. IMPORTANT: tiles are assumed to
+        be in the Spherical Mercator projection (EPSG:3857). [Default is
+        ``xyzservices.providers.Stamen.Terrain``, i.e. Stamen Terrain web
+        tiles].
+
+    lonlat : bool
+        Optional. If False, coordinates in ``region`` are assumed to be
+        Spherical Mercator as opposed to lon/lat. [Default is True].
+
+    wait : int
+        Optional. If the tile API is rate-limited, the number of seconds to
+        wait between a failed request and the next try. [Default is 0].
+
+    max_retries : int
+        Optional. Total number of rejected requests allowed before contextily
+        will stop trying to fetch more tiles from a rate-limited API. [Default
+        is 2].
+
+    kwargs : dict
+        Extra keyword arguments to pass to :func:`contextily.bounds2img`.
+
+    Returns
+    -------
+    raster : xarray.DataArray
+        Georefenced 3D data array of RGB value.
+
+    Raises
+    ------
+    ModuleNotFoundError
+        If ``contextily`` is not installed. Follow
+        :doc:`install instructions for contextily <contextily:index>`, (e.g.
+        via ``pip install contextily``) before using this function.
+
+    Examples
+    --------
+    >>> import contextily
+    >>> from pygmt.datasets import load_tile_map
+    >>> raster = load_tile_map(
+    ...     region=[103.60, 104.06, 1.22, 1.49],  # West, East, South, North
+    ...     source=contextily.providers.Stamen.TerrainBackground,
+    ...     lonlat=True,  # bounding box coordinates are longitude/latitude
```

Comment:
Remove this line because it's the default option?
```suggestion
```

Reply:
I think it's ok to keep it in the example just to be more explicit.